packages/filebrowser-extension/src/index.ts

```diff
@@ -232,6 +230,23 @@ function activateBrowser(app: JupyterLab, factory: IFileBrowserFactory, restorer
  * Add the main file browser commands to the application's command registry.
  */
 function addCommands(app: JupyterLab, tracker: InstanceTracker<FileBrowser>, browser: FileBrowser): void {
+  const getBrowserForPath = (path: string): FileBrowser => {
+    const driveName = app.serviceManager.contents.driveName(path);
+
+    if (driveName) {
+      let browserForPath = tracker.find((fb: FileBrowser) => fb.id.toLowerCase().startsWith(driveName.toLowerCase()));
```

Comment:
It definitely would be cleaner to match path drive to filebrowser drive directly, rather than in the current roundabout way through `fb.id`. I like it